apps/emqx_durable_storage/src/emqx_ds_lts.erl

```diff
@@ -542,9 +542,19 @@ do_topic_key(Trie, _, _, State, [], _Parent, Tokens, Varying) ->
 do_topic_key(Trie, ThresholdFun, Depth, State, [Tok | Rest], Parent, Tokens, Varying0) ->
     % TODO: it's not necessary to call it every time.
     Threshold = ThresholdFun(Depth, Parent),
-    {NChildren, IsWildcard, NextState} = trie_next_(Trie, State, Tok),
+    %% If the threshold is 0, we do not want to create a new static node for this level.
+    CreateTok =
+        case Threshold of
+            0 -> ?PLUS;
+            _ -> Tok
+        end,
+    {NChildren, IsWildcard, NextState} = trie_next_(Trie, State, Tok, CreateTok),
```

Comment:
Looks a bit ugly, but the logic here is quite critical & complex, so I tried not to change it significantly.